packages/langium/src/serializer/hydrator.ts

```diff
@@ -12,25 +12,26 @@ import { isAbstractElement, type AbstractElement, type Grammar } from '../langua
 import type { Linker } from '../references/linker.js';
 import type { Lexer } from '../parser/lexer.js';
 import type { LangiumCoreServices } from '../services.js';
-import type { Reference, AstNode, CstNode, LeafCstNode, GenericAstNode, Mutable } from '../syntax-tree.js';
+import type { ParseResult } from '../parser/langium-parser.js';
+import type { Reference, AstNode, CstNode, LeafCstNode, GenericAstNode, Mutable, RootCstNode } from '../syntax-tree.js';
 import { isRootCstNode, isCompositeCstNode, isLeafCstNode, isAstNode, isReference } from '../syntax-tree.js';
 import { streamAst } from '../utils/ast-utils.js';
 import { BiMap } from '../utils/collections.js';
 import { streamCst } from '../utils/cst-utils.js';
 
 /**
- * The hydrator service is responsible for allowing AST nodes to be sent across worker threads.
+ * The hydrator service is responsible for allowing AST parse results to be sent across worker threads.
  */
 export interface Hydrator {
     /**
-     * Converts an AST node to a plain object. The resulting object can be sent across worker threads.
+     * Converts parse result to a plain object. The resulting object can be sent across worker threads.
      */
-    dehydrate(node: AstNode): object;
+    dehydrate(result: ParseResult<AstNode>): object;
     /**
-     * Converts a plain object to an AST node. The resulting AST node can be used in the main thread.
-     * Calling this method on non-plain objects will result in undefined behavior.
+     * Converts a plain object to a parse result. The resulting AST node can be used in the main thread.
+     * Calling this method on objects that have not been dehydrated first will result in undefined behavior.
      */
-    hydrate(node: object): AstNode;
+    hydrate<T extends AstNode = AstNode>(result: object): ParseResult<T>;
```

Comment:
I like types. You are expecting a certain structure down below. This structure should be reflected by a type definition and not by `object`.

```ts
interface {
  lexerErrors: LexerError[],
  parserErrors: parserError[],
  value: object //for this guy you are not able to assume a type
};
```
WDYT?

Reply:
Yep, thanks. That's indeed better 👍